zabbixsender/zabbixsvc/zabbixsvc_test.go

```diff
@@ -151,122 +139,99 @@ func TestJSONHandlerOK(t *testing.T) {
 
 	go func() {
 		for {
-			log.Info("here")
 			conn, err := l.Accept()
 			if err != nil {
 				t.Fatal(err)
 			}
 			defer conn.Close()
-			buf := make([]byte, 112)
+			buf := make([]byte, 141)
 			_, err = conn.Read(buf)
 			if err != nil {
 				t.Fatal(err)
 			}
 			log.Info(string(buf))
 
-			if msg := string(buf); msg != "ZBXD\x01x\x00\x00\x00\x00\x00\x00\x00{\"request\":\"sender data\",\"data\":[{\"host\":\"Testing\",\"key\":\"any.instancedown\",\"value\":\"0\",\"clock\":48}" {
-				t.Fatalf("Unexpected message:\nGot:\t\t%s\nExpected:\t%s\n", msg, "Request sent")
+			if msg := string(buf[:105]); msg != expectedMsg {
+				t.Fatalf("Unexpected message:\nGot:\t\t%s\nExpected:\t%s\n", msg, expectedMsg)
 			}
 			_, err = conn.Write([]byte("ZBXD\x01Z\x00\x00\x00\x00\x00\x00\x00{\"response\":\"success\",\"info\":\"processed: 1; failed: 0; total: 1; seconds spent: 0.000041\"}"))
 			if err != nil {
 				t.Fatal(err)
 			}
 
-			conn.Read(buf)
-
```

Comment:
So we get 141 bytes and if we compare only 105 ones(that we actually need) we don't need to read the leftovers, due to the fact that there are none, clever.